src/app/profiles/store/sagas/windowConnection.saga.ts

```diff
@@ -0,0 +1,55 @@
+import { channel } from 'redux-saga';
+import {
+  put,
+  call,
+  delay,
+  fork,
+  takeEvery,
+  select,
+  actionChannel
+} from 'redux-saga/effects';
+import { AppAction } from 'app/actions';
+import { connect } from 'app/store/actions/connection';
+import stripReceiverMeta from 'app/store/stripReceiverMeta';
+import watchWindow from 'app/store/sagas/window/watch.saga';
+import extensionId from 'app/profiles/extensionId';
+import { isConnected } from 'app/profiles/store/selectors/connection';
+import createMessageSender from 'webext/createMessageSender';
+
+function* putAndStripActionSaga(action: AppAction) {
+  yield put(stripReceiverMeta(action));
+}
+
+function* attemptToConnectPeriodicallySaga(targetOrigin = '*') {
+  const incomingChannel = yield call(channel);
+  const outgoingChannel = yield actionChannel(
+    ({ meta }: AppAction) => meta?.receiver?.id === extensionId
+  );
+  yield fork(
+    watchWindow,
+    window,
+    targetOrigin,
+    incomingChannel,
+    outgoingChannel
+  );
+  yield takeEvery(incomingChannel, putAndStripActionSaga);
+
+  const attemptInterval = 5000;
+  if (attemptInterval > 0) {
+    while (true) {
+      const connected = yield select(isConnected);
+      if (!connected) {
+        yield put(
+          connect(undefined, {
+            receiver: createMessageSender({ id: extensionId })
+          })
+        );
+      }
+      yield delay(attemptInterval);
+    }
+  }
+}
+
+export default function* windowConnectionSaga(targetOrigin = '*') {
```

Comment:
I suggest we could just `export default attemptToConnectPeriodicallySaga` here … ?